pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ApplicationLifecycle.kt

```diff
@@ -21,30 +21,47 @@ class AppActivityLifecycleCallbacks: Application.ActivityLifecycleCallbacks {
     var startedCount = 0
     var stoppedCount = 0
     fun appInBackground(): Boolean = startedCount <= stoppedCount
+
+    var currentActivity: WeakReference<Activity>? = null
```

Comment:
Maybe we can make this `internal`

Reply:
If we are using weak references in multiple places, and to avoid doing everywhere `ref?.get()` we can have a delegate defined like:

```
fun <T> weakReference() = object : ReadWriteProperty<T?> {
    private var reference: WeakReference<T>? = null
    operator fun getValue(thisRef: Any, property: KProperty<*>): T? =
        reference?.get()
    operator fun setValue(thisRef: Any, property: KProperty<*>, value: T?) {
        reference = value?.let { WeakReference(it) }
    }
}
```
And then you can just define this like:
```
internal var currentActivity by weakReference<Activity>()
```

And it'll be safe to assign or get without having to deal with the weak reference interface

Reply:
I think this looks nice, but I think it's a bit too much faff and I don't want to encourage more WeakReference-related code.